akka-http-core/src/main/java/akka/http/javadsl/model/headers/ModeledCustomHeader.java

```diff
@@ -0,0 +1,43 @@
+package akka.http.javadsl.model.headers;
+
+import akka.http.impl.util.Rendering;
+import akka.util.Helpers;
+
+/**
+ * Support class for building user-defined custom headers defined by implementing `name` and `value`.
+ * By implementing a {@link ModeledCustomHeader} along with {@link ModeledCustomHeaderCompanion} instead of {@link CustomHeader} directly,
+ * convenience methods that allow parsing this user-defined header from {@link akka.http.javadsl.model.HttpHeader}  are
+ * available to use.
+ */
+public abstract class ModeledCustomHeader extends CustomHeader {
+
+  private final String name;
+  private final String value;
+
+  protected ModeledCustomHeader(final String name, final String value) {
+    super();
+    this.name = name;
+    this.value = value;
+  }
+
+  @Override
+  public String name() {
+    return name;
+  }
+
+  @Override
+  public String lowercaseName() {
+    return Helpers.toRootLowerCase(name);
+  }
+
+  @Override
+  public String value() {
+    return value;
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public <R extends Rendering> R render(R r) {
+    return (R) r.$tilde$tilde(name).$tilde$tilde(':').$tilde$tilde(' ').$tilde$tilde(value);
```

Comment:
can't we have the `Rendering` trait expose more java-friendly method alternatives?